### PR TITLE
ensure that duplicate keys do not overlap

### PR DESCRIPTION
**Pull request by matejcik**

### CODE REVIEW

**Comment by matejcik** [comment]
@techi602 please check this out

**Comment by techi602** [comment]
Good, now I see coingeckoId: `shiba-inu` in the data and with ID starting erc20 so that should solve our issue. 

image

But I am wondering why bitcoin does not have coingeckoId:

image

It is not big issue, the shop will pair the coin using coin name. I am just curious.

Please let me try to run the import on our devel server first, I will give you feedback afterwards

**Comment by matejcik** [comment]
> But I am wondering why bitcoin does not have coingeckoId:

we don't fetch bitcoin data from coingecko.
we can hard-code some coins in the overrides file (please provide a PR to `coins_details.overrides.json` if you have more that you would like), or let the pairing up to you

File: coins_details/coins_details.py [hunk 1]
```diff
@@ -132,39 +131,39 @@ def update_simple(
...
 
     return res
 
 
-def update_nem_mosaics(coins: Coins, support_info: SupportInfo) -> dict[str, Coin]:
+def update_nem_mosaics(coins: Coins, support_info: SupportInfo) -> Coins:
     res = update_simple(coins, support_info, "mosaic")
-    for coin in coins:
-        key = coin["key"]
+    for coin, updated in zip(coins, res):
```
Comment (grdddj) [inline coins_details/coins_details.py:156]: Nit: `coin` is not used here

**Review by grdddj — APPROVED** [review]
LGTM, just one minor code smell

**Review by techi602 — APPROVED** [review]

**Comment by techi602** [comment]
Thanks, tests passed, waiting for merge.

File: coins_details/coins_details.py (hunk 1 above)
Comment (matejcik) [reply coins_details/coins_details.py]: fixed in force-push

**Pull request merged**